harvester/sources/extraction/edurep.py

```diff
@@ -0,0 +1,286 @@
+import re
+from mimetypes import guess_type
+from hashlib import sha1
+from core.constants import HIGHER_EDUCATION_LEVELS
+from django.conf import settings
+from dateutil.parser import parse as date_parser
+from datagrowth.processors import ExtractProcessor
+from django.utils.text import slugify
+
+
+FILE_TYPE_TO_MIME_TYPE = {
+    "TEXT": "application/pdf",
+    "VIDEO": "video/mp4",
+    "AUDIO": "audio/mp3",
+    "IMAGE": "image/jpeg",
+    "DOCUMENT": "application/x-Wikiwijs-Arrangement",
+    "WEBSITE": "text/html",
+    "NOT_SET": None
+}
+
+
+class EdurepMetadataExtraction(ExtractProcessor):
+
+    youtube_regex = re.compile(r".*(youtube\.com|youtu\.be).*", re.IGNORECASE)
+    cc_url_regex = re.compile(r"^https?://creativecommons\.org/(?P<type>\w+)/(?P<license>[a-z\-]+)/(?P<version>\d\.\d)",
+                              re.IGNORECASE)
+    cc_code_regex = re.compile(r"^cc([ \-][a-z]{2})+$", re.IGNORECASE)
+
+    @classmethod
+    def get_record_state(cls, node):
+        return "active"
+
+    #############################
+    # GENERIC
+    #############################
+
+    @staticmethod
+    def _serialize_access_rights(access_rights):
+        access_rights = access_rights.replace("Access", "")
+        access_rights = access_rights.lower()
+        access_rights += "-access"
+        return access_rights
+
+    @classmethod
+    def get_files(cls, node):
+        if not node:
+            return []
+        return {
+            "title": node["schema:name"]["@value"],
+            "url": node["schema:url"],
+            "mime_type": FILE_TYPE_TO_MIME_TYPE.get(node["schema:encodingFormat"]),
+            "hash": sha1(node["schema:url"].encode("utf-8")).hexdigest(),
+            "copyright": node["schema:license"],
+            "access_rights": node["dcterms:accessRights"]
+        }
+
+    @classmethod
+    def get_language(cls, node):
+        language = node["schema:identifier"]["schema:inLanguage"]
+        if language == "Dutch":
+            return "nl"
+        elif language == "English":
+            return "en"
+        return "unk"
+
+    @classmethod
+    def get_url(cls, node):
+        files = cls.get_files(node)
+        if not files:
+            return node["portalUrl"]
```

Comment:
Maybe a node.get call with None as default is safer than an item access without any error handling. We'll find out if we run this on all 2000+ materials, but I'm expecting some crashes here. Let's buy some popcorn and watch :partying_face: